posixlib/src/main/resources/scala-native/sys/socket.c

```diff
@@ -196,8 +214,28 @@ ssize_t scalanative_recv(int socket, void *buffer, size_t length, int flags) {
     return recv(socket, buffer, length, flags);
 }
 
-ssize_t scalanative_send(int socket, void *buffer, size_t length, int flags) {
+ssize_t scalanative_recvfrom(int socket, void *buffer, size_t length, int flags,
+                             struct scalanative_sockaddr *address,
+                             socklen_t *address_len) {
+    // Using this function rather than a direct call in socket.scala allows
+    // _Static_asserts at the top of this file to check structure requirements.
+
+    return recvfrom(socket, buffer, length, flags, (struct sockaddr *)address,
+                    address_len);
+}
+
+int scalanative_send(int socket, void *buffer, size_t length, int flags) {
     return send(socket, buffer, length, flags);
 }
 
+ssize_t scalanative_sendto(int socket, void *buffer, size_t length, int flags,
+                           struct scalanative_sockaddr *address,
+                           socklen_t address_len) {
+    // Using this function rather than a direct call in socket.scala allows
```

Comment:
Is it really needed? I've tried it locally with direct calls and modified condition to force failure and it works correctly. 
```
[error] ..//sys/socket.c:21:1: error: static_assert failed due to requirement '__builtin_offsetof(struct scalanative_sockaddr, sa_data) == 1' "Unexpected size: scalanative_sockaddr sa_family"
[error] _Static_assert(offsetof(struct scalanative_sockaddr, sa_data) == 1,
```

Tested on Ubuntu 20.04, clang 10.0

Reply:
You are absolutely correct, given the assurances at the top of `socket.c`.  

Because of prior art in `socket.scala`, there is some room for discussion/uncertainty
about how to do the steps of the evolution dance.  

I took the conservative approach, of making the new methods indirect, as are all 
the other methods in `socket.scala`. This went against my drive for conceptual
simplicity and runtime efficiency but I thought this would lead to less confusion.

Given your insightful comment, I will try going the other way around, both
for this and the followup `sendmsg/recvmsg` scatter/gather PR.

Eventually, over an unfortunately long series of PRs, I hope to make
all of the socket calls direct. This requires community acceptance of the
concept of using the C compiler to ensure structure matches (rather
than runtime copy_in/copy_out).  It also requires building up a series
of _Static_assert statemets in `socket.c` and ensuring that it continues
to get compiled.